disnake/role.py

```diff
@@ -536,14 +602,39 @@ async def edit(
             await self._move(position, reason=reason)
 
         payload: Dict[str, Any] = {}
-        if color is not MISSING:
-            colour = color
 
-        if colour is not MISSING:
-            if isinstance(colour, int):
-                payload["color"] = colour
+        colors: Dict[str, Any] = {}
+        if color is not MISSING or colour is not MISSING:
+            primary_colour = color or colour
+
+        if primary_color is not MISSING:
+            primary_colour = color
+
+        if primary_colour is not MISSING:
+            if isinstance(primary_colour, int):
+                colors["primary_color"] = primary_colour
             else:
-                payload["color"] = colour.value
+                colors["primary_color"] = primary_colour.value
+
+        if secondary_color is not MISSING:
+            secondary_colour = secondary_color
+
+        if secondary_colour is not MISSING:
+            if isinstance(secondary_colour, Colour):
+                colors["secondary_color"] = secondary_colour.value
+            else:
+                colors["secondary_color"] = secondary_colour
+
+        if tertiary_color is not MISSING:
+            tertiary_colour = tertiary_color
+
+        if tertiary_colour is not MISSING:
+            if isinstance(tertiary_colour, Colour):
+                colors["tertiary_color"] = tertiary_colour.value
+            else:
+                colors["tertiary_color"] = tertiary_colour
+
+        payload["colors"] = colors
```

Comment:
This currently sends `"colors": {}` when colors are not being changed, I think to be safe we should only be sending it when necessary.

(see also https://github.com/DisnakeDev/disnake/pull/1312/files#r2314431454, which I believe would resolve this as a side effect as well)

Reply:
https://github.com/DisnakeDev/disnake/pull/1312/commits/8db0056691912da83b96ca5504e1b9d79501aecf